### ISSUE TITLE
Renaming origin remote doesn't seem to remove it from git/config

### ISSUE THREAD

**Issue by msuozzo** [martinvonz/jj#3876]
## Steps to Reproduce the Problem

1. jj git clone $REPO
2. jj remote add up $UPSTREAM
3. jj remote rename origin og
4. jj commit -m "Foo"
5. jj push --tracked

## Expected Behavior

Successful push

`.jj/repo/store/git/config`:

```
[remote "up"]
        url = ...
        fetch = +refs/heads/*:refs/remotes/up/*
[remote "og"]
        url = ...
        fetch = +refs/heads/*:refs/remotes/og/*
```

## Actual Behavior

```
Creating branch push-pquuyxwzzynr for revision this
Branch changes to push to origin:
  Add branch push-pquuyxwzzynr to 2311786c588b
Error: No git remote named 'origin'
```

`.jj/repo/store/git/config` (contains an additional `[remote "origin"]`):

```
[remote "origin"]
[remote "up"]
        url = ...
        fetch = +refs/heads/*:refs/remotes/up/*
[remote "og"]
        url = ...
        fetch = +refs/heads/*:refs/remotes/og/*
```

## Specifications

- Platform: Debian
- Version: 0.18.0

**Comment by yuja** [comment]
.git/config is managed by git2/libgit2.

Does the empty `[remote "origin"]` section matter? `git config -l` appears to ignore it.

**Comment by msuozzo** [comment]
Yeah feel free to close. I had thought there was an issue related to that but it was just with `jj git fetch/push` but i just had to update the default config values. Apropos of that, though, it'd be nice if jj could provide a warning when a remote in one of those `git.push` and `git.fetch` defaults are removed/renamed.
